sdk/core/core/src/az_http_response_parser.c

```diff
@@ -189,54 +203,18 @@ AZ_NODISCARD az_result az_http_response_parser_read_header(
   return AZ_OK;
 }
 
-AZ_NODISCARD az_result az_http_response_parser_read_body(
+AZ_NODISCARD az_result az_http_response_parser_get_body(
     az_http_response_parser * const self,
     az_http_response_body * const out) {
   AZ_CONTRACT_ARG_NOT_NULL(self);
   AZ_CONTRACT_ARG_NOT_NULL(out);
 
+  if (self->kind != AZ_HTTP_RESPONSE_BODY) {
+    return AZ_HTTP_ERROR_INVALID_STATE;
```

Comment:
I see that you have this here but your design means a run-time failure while my design is more of a compile-time error since you had to pass the &lastHeader in as an argument to this method.

Reply:
What will happen if someone pass not the last header to your function?

Reply:
Yes, this is a potential problem but I think it is less likely a customer will make the error because the API requires a header in order to call it. Your design does not have this requirement and so it is easy for a customer to think they can just get the body without enumerating through the headers.